src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -17,13 +17,27 @@ import io.circe.derivation.deriveDecoder
 import scala.concurrent.duration._
 import scala.util.control.NonFatal
 
+class AuthSttpBackend[R[_], -S](delegate: SttpBackend[R, S], authProvider: AuthProvider[R])
+    extends SttpBackend[R, S] {
+  override def send[T](request: Request[T, S]): R[Response[T]] =
+    responseMonad.flatMap(authProvider.getAuth) { auth: Auth =>
```

Comment:
https://ci.jenkins.cognite.ai/blue/rest/organizations/jenkins/pipelines/cognitedata-ci/pipelines/relationships-api/branches/PR-950/runs/1/nodes/54/steps/63/log/?start=0
```
[info] CdfClientProviderTest:
[info] AuthTokenProvider
[info] - should return bearer token *** FAILED *** (3 milliseconds)
[info]   java.lang.NullPointerException:
[info]   at com.cognite.sdk.scala.v1.AuthSttpBackend.send(Client.scala:23)
[info]   at com.softwaremill.sttp.RequestT.send(RequestT.scala:246)
[info]   at com.cognite.sdk.scala.v1.RequestSession.get(Client.scala:137)
[info]   at com.cognite.sdk.scala.common.Login.status(login.scala:28)
[info]   at com.cognite.relationships.api.common.service.sdk.SdkClientGenerator.makeGenericClient(SdkClientGenerator.scala:35)
[info]   at com.cognite.relationships.api.common.service.sdk.SdkClientGenerator.makeGenericClient$(SdkClientGenerator.scala:16)
[info]   at com.cognite.relationships.api.common.service.sdk.CdfClientProvider$.makeGenericClient(CdfClientProvider.scala:14)
[info]   at com.cognite.relationships.api.common.service.sdk.CdfClientProvider$.$anonfun$getTokenAuthClient$3(CdfClientProvider.scala:39)
[info]   at cats.data.EitherT.$anonfun$flatMap$1(EitherT.scala:405)
[info]   at flatMap @ com.cognite.relationships.api.common.service.http.OpaqueCursorSecretVerifier.verify(OpaqueCursorSecretVerifier.scala:23)
```
unit tests start throwing NPE in relationships-api, any hints on what could be the reason?

Reply:
Which sttpBackend are you guys using?

Reply:
could be stub in tests https://github.com/cognitedata/relationships-api/blob/master/api/src/test/scala/com/cognite/relationships/api/common/service/sdk/CdfClientProviderTest.scala#L44